### DISCUSSION TITLE
how to identify the different types of data links from a GranuleQuery

### DISCUSSION THREAD

**Discussion by tjhearty** [nasa/python_cmr#56] [Q&A]
How can I get a list of data, S3, or opendap urls from a GranuleQuery?  

I currently do the following to get the different types of URLs:

# First, get the list of all entry URL links returned by granule metadata
all_links = [entry['links'] for entry in gran_met]

# Next, get just the granule URL links, exclude inherited links (i.e. from collection metadata).
gran_links = [link for group in all_links for link in group if 'inherited' not in link.keys()]

From there I can search for string in the 'title' or 'rel' to identify the data, S3, or opendap urls?  Is there another way to get a list of the different types of urls?

**Comment by chuckwondo** [comment]
@tjhearty, apologies for taking so long to answer your question. At the moment, there are not convenience functions/methods available for this. However, you may wish to look at the [earthaccess](https://github.com/nsidc/earthaccess) library, which makes use of `python-cmr`, and does have some convenience methods that may work for you. In particular, the [data_links](https://earthaccess.readthedocs.io/en/latest/user-reference/granules/granules/#earthaccess.results.DataGranule.data_links) method may be of interest to you, which is a method of the granule objects returned from a granule query.
